### PR TITLE
Add mock acceptance test for git-repository resource

### PR DESCRIPTION
**Pull request by Shubbhang351**
1. **What this PR does / why we need it**:

Add mock acceptance test for git-repository resource

Git Repository mock Accep. test

Screenshot 2023-07-15 at 2 45 11 PM

Git Repository Accep. test

Screenshot 2023-07-17 at 4 11 46 PM

Screenshot 2023-07-15 at 2 46 25 PM

### CODE REVIEW

**Comment by ishangupta-ds** [comment]
@Shubbhang351 please incorporate changes as per reviews on - https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/187

**Comment by Shubbhang351** [comment]
acceptance test

Screenshot 2023-07-15 at 2 46 25 PM

File: internal/resources/gitrepository/resource_git_repository_test.go [hunk 1]
```diff
@@ -55,105 +58,178 @@ func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
...
+
+		log.Println("Setting up the mock endpoints...")
+		testConfig.setupHTTPMocks(t)
+	} else {
+		// Environment variables with non default values required for a successful call to MKP
+		requiredVars := []string{
+			"VMW_CLOUD_ENDPOINT",
+			"TMC_ENDPOINT",
+			"VMW_CLOUD_API_TOKEN",
+			"ORG_ID",
```
Comment (ramya-bangera) [inline internal/resources/gitrepository/resource_git_repository_test.go:100]: Is ORG_ID required param for this resource?

File: internal/resources/gitrepository/resource_git_repository_test.go [hunk 2]
```diff
@@ -55,105 +58,178 @@ func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
...
+	if !found {
+		os.Setenv("TF_ACC", "true")
+		os.Setenv("TMC_ENDPOINT", "dummy.tmc.mock.vmware.com")
+		os.Setenv("VMW_CLOUD_API_TOKEN", "dummy")
+		os.Setenv("VMW_CLOUD_ENDPOINT", "console.cloud.vmware.com")
+
+		log.Println("Setting up the mock endpoints...")
+		testConfig.setupHTTPMocks(t)
+	} else {
+		// Environment variables with non default values required for a successful call to MKP
```
Comment (ramya-bangera) [inline internal/resources/gitrepository/resource_git_repository_test.go:95]: Please run linters manually before sending out for review as these checkins are to feature branch and not main. lint: full stop at the end,

File: internal/resources/gitrepository/resource_git_repository_test.go [hunk 3]
```diff
@@ -55,105 +58,178 @@ func testGetDefaultAcceptanceConfig(t *testing.T) *testAcceptanceConfig {
...
+
 	t.Log("start git repository resource acceptance tests!")
 
-	// Test case for git repository resource for GO_GIT git implementation type.
+	// Test case for git repository resource.
 	resource.Test(t, resource.TestCase{
 		PreCheck:          testhelper.TestPreCheck(t),
 		ProviderFactories: testhelper.GetTestProviderFactories(testConfig.Provider),
-		CheckDestroy:      nil,
+		// CheckDestroy:      nil,
```
Comment (ramya-bangera) [inline internal/resources/gitrepository/resource_git_repository_test.go:117]: Remove this commented line

**Review by ramya-bangera — COMMENTED** [review]
LGTM

File: internal/resources/gitrepository/resource_git_repository_test.go (hunk 3 above)
Comment (Shubbhang351) [reply internal/resources/gitrepository/resource_git_repository_test.go]: updated PTAL

File: internal/resources/gitrepository/resource_git_repository_test.go (hunk 2 above)
Comment (Shubbhang351) [reply internal/resources/gitrepository/resource_git_repository_test.go]: I've tried but didn't get any linter issues might need to bump up the version.
Updated PTAL

File: internal/resources/gitrepository/resource_git_repository_mock_test.go [hunk 4]
```diff
@@ -0,0 +1,644 @@
...
+	}
+
+	postCGContinuousDeliveryResponse := &continuousdeliveryclustergroupmodel.VmwareTanzuManageV1alpha1ClustergroupFluxcdContinuousdeliveryContinuousDeliveryResponse{
+		ContinuousDelivery: postResponseCDContinuousDelivery,
+	}
+
+	return postCGRequest, postCGResponse, getCGResponse, postCGContinuousDeliveryRequest, postCGContinuousDeliveryResponse
+}
+
+func (testConfig *testAcceptanceConfig) getClRequest(orgID string, referenceArray []*objectmetamodel.VmwareTanzuCoreV1alpha1ObjectReference) (
```
[the remark this answers is not in the document]
Comment (Shubbhang351) [reply internal/resources/gitrepository/resource_git_repository_mock_test.go]: updated

File: internal/resources/gitrepository/resource_git_repository_test.go (hunk 1 above)
Comment (Shubbhang351) [reply internal/resources/gitrepository/resource_git_repository_test.go:100]: for resource orgid is opeional field, this one is for Acceptance test to make successful call to service

**Review by ishangupta-ds — APPROVED** [review]
LGTM.

**Review by ramya-bangera — APPROVED** [review]

**Pull request merged**